script.js

```diff
@@ -1 +1,464 @@
+    const currentSuitSymbol = suits[suitIndex][1];
+    const currentCardColor = suits[suitIndex][2];
+    // console.log(`current suit: ${currentSuit}`);
+
+    // loop to create all cards in this suit
+    // rank 1-13
+    for (let rankCounter = 1; rankCounter <= 13; rankCounter += 1) {
+      // Convert rankCounter to string
+      let cardName = `${rankCounter}`;
+      let displayName = cardName;
+
+      // 1, 11, 12 ,13
+      if (cardName === '1') {
+        cardName = 'ace';
+        displayName = 'A';
+      } else if (cardName === '11') {
+        cardName = 'jack';
+        displayName = 'J';
+      } else if (cardName === '12') {
+        cardName = 'queen';
+        displayName = 'Q';
+      } else if (cardName === '13') {
+        cardName = 'king';
+        displayName = 'K';
+      }
+
+      // make a single card object variable
+      const card = {
+        name: cardName,
+        display: displayName,
+        suit: currentSuit,
+        suitSymbol: currentSuitSymbol,
+        rank: rankCounter,
+        color: currentCardColor,
+        matched: false,
+      };
+
+      // console.log(`rank: ${rankCounter}`);
+
+      // add the card to the deck
+      newDeck.push(card);
+      // Cards are duplicated - for matching cards
+      newDeck.push(card);
+    }
+  }
+  return newDeck;
+};
+
+// Used to store all the cards of desired size in a 2D manner
+const boardOfCards = [];
+let firstCard = null;
+const boardSize = 4; // has to be an even number
+const delayForGameStatusInMilliSeconds = 3 * 1000;
+const delayForSpecialMessage = 5 * 1000;
+const intervalSingleSecond = 1000;
+const intervalInMinutes = 1;
+const intervalInSeconds = intervalInMinutes * 60;
+const intervalInMilliSeconds = 1000 * intervalInSeconds; // 180000 = 3 minutes
+let countDownTime = intervalInMilliSeconds;
+// Store the cardElement of the first card.
+// Used to turn it over, in case a match is not founded.
+let firstCardElement = null;
+let boardElement = null;
+let gameStarted = false;
+// this variable stores the sliced card deck as per the board size
+let deck;
+// To store the reference to interval function to start the game
+const intervalReference = null;
+// To store the reference to the timeout function
+let timeoutIntervalReference = null;
+// To store the name
+let playerName = '';
+// // Score variable
+let gameTotalScore = 0;
+let gameAttempts = 0;
+
+// A div element to display information on Game status when necessary
+const divGameStatusInfo = document.createElement('div');
+// Input element for entering the player name
+const inputPlayerName = document.createElement('input');
+// Button to submit the name
+const inputNameSubmitButton = document.createElement('button');
+// Refresh button
+const resetButton = document.createElement('button');
+// element for displaying countdown timer
+const divTimer = document.createElement('div');
+// create a p element for displaying time
+const paraTimerElement = document.createElement('p');
```

Comment:
Ok to declare globals on top, but consider initialising these elements in `gameInit` so that the DOM initialisation code is centralised in 1 place.